pkg/openstack/loadbalancer.go

```diff
@@ -563,14 +564,24 @@ func (lbaas *LbaasV2) deleteOctaviaListeners(lbID string, listenerList []listene
 	return nil
 }
 
-func (lbaas *LbaasV2) createFloatingIP(msg string, floatIPOpts floatingips.CreateOpts) (*floatingips.FloatingIP, error) {
+func (lbaas *LbaasV2) createFloatingIP(msg string, floatIPOpts floatingips.CreateOpts, tag *string) (*floatingips.FloatingIP, error) {
 	klog.V(4).Infof("%s floating ip with opts %+v", msg, floatIPOpts)
 	mc := metrics.NewMetricContext("floating_ip", "create")
 	floatIP, err := floatingips.Create(lbaas.network, floatIPOpts).Extract()
 	err = PreserveGopherError(err)
 	if mc.ObserveRequest(err) != nil {
 		return floatIP, fmt.Errorf("error creating LB floatingip: %v", err)
 	}
+
+	if tag != nil {
+		err = attributestags.Add(lbaas.network, "floatingips", floatIP.ID, *tag).ExtractErr()
+		err = PreserveGopherError(err)
+		if mc.ObserveRequest(err) != nil {
+			floatingips.Delete(lbaas.network, floatIP.ID)
```

Comment:
I'm not really sure a failure to add a tag should be fatal and cause a full rollback. This most likely won't work either. I'd rather have the code follow K8s reconciling principles and make it fetch the FIP, check if tag exists and if not, add it. This would also handle the upgrade situations, so that after upgrade all FIPs without tags would get one.

Reply:
sure ，this is indeed not very clean. Now it is modified to check the tag when confirming the floating ip. If it does not exist, it will be added to the floating ip.